### DISCUSSION TITLE
Labels for locales

### DISCUSSION THREAD

**Discussion by qbeauperin** [payloadcms/payload#2702] [Feature Requests & Ideas]
> _Following up on a [Discord discussion](https://discord.com/channels/967097582721572934/1091805251344019609/1098263056926117898)._ with @DanRibbens 

Having the ability to set labels for our locales would help making the admin locale picker a lot more user friendly and noticeable.

Example, before ➡️ after:
![image](https://github.com/payloadcms/payload/assets/1763338/5cf611b6-d514-4e34-a5c4-30d11a6f36a8) ➡️ ![image](https://github.com/payloadcms/payload/assets/1763338/fc00b8da-213e-4f90-a2a8-a6c47b10a0b8)

---

In terms of implementation, this would extend the current config by accepting either a string (current logic) or an object.

For example, this:
```typescript
localization: {
    locales: ['en', 'fr']
}
```

could become this:
```typescript
localization: {
    locales: [
        {
            value: 'en',
            label: 'English', // or i18n object for admin i18n
        },
        {
            value: 'fr',
            label: 'French', // or i18n object for admin i18n
        },
    ]
}
```

**Comment by qbeauperin** [comment]
Oh wow, looks like @ibr-hin95 might have [already implemented it](https://github.com/payloadcms/payload/commit/c86a9677ec4a03a256e55b5d68a4fab79dfe1523) within their "Arabic and RTL support" PR (#2657)! 😮🥳

**Reply by DanRibbens** [reply]
Yes! That PR is looking really good. I haven't had time to pull it down and test yet, but I think it's got it covered already.
